kong/templates/nginx_kong.lua

```diff
@@ -23,6 +23,10 @@ lua_shared_dict kong_db_cache               ${{MEM_CACHE_SIZE}};
 lua_shared_dict kong_db_cache_miss          12m;
 lua_shared_dict kong_secrets                5m;
 
+> if not legacy_dns_client then
+lua_shared_dict kong_dns_cache              12m;
```

Comment:
```suggestion
lua_shared_dict kong_dns_cache              5m;
```

I suggest start with something much smaller to reduce the memory impact, and make it configureable via `kong.conf`. Most of the customer have only a handful of domain names despite much larger number of services.

Also, have we ran any tests on how much records xMB of zone can hold? We should document it.

Reply:
done, memory consumptions: https://github.com/Kong/kong/blob/refactor/dns_client/kong/dns/README.md#memory